### PR TITLE
Refactor boolean prop names for consistency

### PR DESCRIPTION
**Pull request by m10l**
# Purpose of PR

This PR renames boolean prop names to provide a consistent API across all components.

Information about our approach to prop naming can be found in the [component principles](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#component-principles) section of the Forma 36 Component Library readme.

Here is a summary of props changed:

| Old prop | New prop | Components affected |
| - | - | - |
| `disabled` | `isDisabled` | Button, CardActions, Checkbox, CheckboxField, ControlledInput, ControlledInputField, DropdownListItem, EditorToolbarButton, IconButton, ModalConfirm, RadioButton, RadioButtonField, Select, Tabs, TextField, TextInput, TextLink, Textarea, ToggleButton |
| `loading` | `isLoading` | Button, EntryCard, ModalConfirm |
| `selected` | `isSelected` | AssetCard, Card, EntryCard, InlineEntryCard, Tabs, ToggleButton |
| `required` | `isRequired` | Checkbox, ControlledInput, ControlledInputField, Form, FormLabel, RadioButton, Select, SelectField, TextField, TextInput, Textarea |
| `checked` | `isChecked` | Checkbox, CheckboxField, ControlledInput, ControlledInputField, FieldGroup, Form, RadioButton, RadioButtonField |
| `error` | `hasError` | TextField, TextInput, Textarea |
| `withDropdown` | `hasDropdown` | CardActions, EditorToolbarButton, IconButton |
| `active` | `isActive` | DropdownListItem, TableSortingLabel, ToggleButton |
| `indicateDropdown` | `hasDropdown` | Button, Dropdown |
| `textarea` | `isTextarea` | Form, TextField |
| `row` | `isRow` | FieldGroup, Form |
| `labelIsLight ` | `hasLightLabel` | CheckboxField, ControlledInputField, RadioButtonField |
| `countCharacters ` | `hasCharacterCount` | TextField |
| `animate ` | `isAnimated ` | SkeletonContainer |

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/forma-36-react-components/src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -29,9 +29,9 @@ storiesOf('Components|Button', module)
           'muted',
         )}
         size={select('size', [undefined, 'small', 'large'], undefined)}
-        loading={boolean('loading', false)}
-        indicateDropdown={boolean('indicateDropdown', false)}
-        disabled={boolean('disabled', false)}
+        isLoading={boolean('isLoading', false)}
+        hasDropdown={boolean('hasDropdown', false)}
+        isDisabled={boolean('isDisabled', false)}
```
Comment (DanweDE) [inline packages/forma-36-react-components/src/components/Button/Button.stories.tsx]: Wondering about `isFullWidth` below... Wouldn't `hasFullWidth` make more sense grammatically?

Comment (m10l) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx]: @DanweDE I'd say no, `isFullWidth` is correct, and quite common. Think "the component is full-width" vs "the component has full-width" - the former (excuse the pun) makes more sense to me here.

File: packages/forma-36-react-components/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -16,12 +16,12 @@ const styles = require('./Button.css');
 
 export type ButtonProps = {
   icon?: IconType;
-  indicateDropdown?: boolean;
+  hasDropdown?: boolean;
```
Comment (suevalov) [inline packages/forma-36-react-components/src/components/Button/Button.tsx]: I'd call it `withDropdownCaret` or something like this

Comment (m10l) [reply packages/forma-36-react-components/src/components/Button/Button.tsx]: Nice, that's a lot clearer - `hasDropdown` is pretty misleading for this one isn't it. I'll update.

Comment (DanweDE) [reply packages/forma-36-react-components/src/components/Button/Button.tsx]: Do you want to add "with" as an allowed prefix for booleans in the component principles or shouldn't it rather be `hasDropdownCaret`?

Comment (m10l) [reply packages/forma-36-react-components/src/components/Button/Button.tsx]: Yup! That's exactly what we went with - the prop is now `hasDropdownCaret` (your comment is on an outdated diff). I'll mark the conversation as resolved

**Pull request merged**